woocommerce/woocommerce
# Refactor section and subsection blocks

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR refactors the section and subsection blocks. Since those components shared code, this PR eliminates duplicated code by creating the component `SectionHeader`.

Closes #44476.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Follow the same testing instructions described in https://github.com/woocommerce/woocommerce/pull/43725

1. Make sure the feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Add the following template modifications to a `mu-plugin`:

```
<?php

use Automattic\WooCommerce\Admin\Features\ProductBlockEditor\ProductTemplates\GroupInterface;

add_action(
	'woocommerce_block_template_area_product-form_after_add_block_general',
	function ( GroupInterface $general_group ) {
		$my_section = $general_group->add_section(
			array(
				'id'         => 'your-prefix-example-section',
				'attributes' => array(
					'title' => __( 'Example section', 'YOUR_TEXT_DOMAIN' ),
				),
			)
		);
		$my_subsection = $my_section->add_subsection(
			array(
				'id'         => 'your-prefix-example-subsection',
				'order' => 10,
                'attributes' => array(
					'title' => __( 'Example subsection', 'YOUR_TEXT_DOMAIN' ),
				),
			)
		);
		$my_subsection->add_block(
			array(
				'id'             => 'your-prefix-example-text-field-1',
				'blockName'      => 'woocommerce/product-text-field',
				'attributes'     => array(
					'property' => 'meta_data.example_text_field_1',
					'label'    => __( 'Example block 2', 'YOUR_TEXT_DOMAIN' ),
				),
			)
		);
	}
);
```
3. Go to `Products > Add New`
4. Verify the `Example section` and `Example subsection` were added correctly.
3. Go to `Products > Add New > Inventory`. This tab has a subsection with the `SKU` and `Track stock quantity for this product` toggle. Same with the Downloads section.
4. Verify everything works and looks as expected.
5. Create a few variations, edit one variation, and verify the inventory fields in the variation are working well.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
